test/pgtap/etl/replace_name_abbreviations.sql

```diff
@@ -0,0 +1,830 @@
+  ('Foobar Bhd', 'Foobar$bhd', NULL),
+  ('Foobar	Bhd', 'Foobar$bhd', 'tab whitespace'),
+  ('foobhdbar', 'foobhdbar', NULL),
+  ('foo bhd bar', 'foo bhd bar', NULL),
+  ('foobhd bar', 'foobhd bar', NULL),
+  ('foo bhdbar', 'foo bhdbar', NULL),
+  ('bhd foobar', 'bhd foobar', NULL),
+  /* \s+bsc$ */
+  ('foobar bsc', 'foobar$bsc', NULL),
+  ('Foobar Bsc', 'Foobar$bsc', NULL),
+  ('foobarbsc', 'foobarbsc', NULL),
+  ('foo bsc bar', 'foo bsc bar', NULL),
+  ('foobsc bar', 'foobsc bar', NULL),
+  ('foo bscbar', 'foo bscbar', NULL),
+  ('foobscbar', 'foobscbar', NULL),
+  ('bsc foobar', 'bsc foobar', NULL),
+  /* \s+bv$ */
+  ('foobar bv', 'foobar$bv', NULL),
+  ('Foobar Bv', 'Foobar$bv', NULL),
+  ('foobarbv', 'foobarbv', NULL),
+  ('foo bv bar', 'foo bv bar', NULL),
+  ('foobv bar', 'foobv bar', NULL),
+  ('foo bvbar', 'foo bvbar', NULL),
+  ('foobvbar', 'foobvbar', NULL),
+  ('bv foobar', 'bv foobar', NULL),
+  ('Foobar Bvi', 'Foobar Bvi', 'Bvi'),
+  ('Foobar Ibv', 'Foobar Ibv', 'Ibv'),
+  /* \s+co$ */
+  ('foobar co', 'foobar$co', NULL),
+  ('Foobar Co', 'Foobar$co', NULL),
+  ('Foobar	Co', 'Foobar$co', 'tab whitespace'),
+  ('foocobar', 'foocobar', NULL),
+  ('foo co bar', 'foo co bar', NULL),
+  /* \s+corp$ */
+  ('foobar corp', 'foobar$corp', NULL),
+  ('Foobar Corp', 'Foobar$corp', NULL),
+  ('Foobar	Corp', 'Foobar$corp', 'tab whitespace'),
+  ('Foocorp', 'Foocorp', 'replacing a suffix needs whitespace'),
+  ('corpFoo', 'corpFoo', NULL),
+  ('Scorpion', 'Scorpion', 'do not replace corp in middle of word'),
+  ('foo corp bar', 'foo corp bar',  'do not replace corp in middle of string'),
+  /* \s+cv$ */
+  ('foobar cv', 'foobar$cv', NULL),
+  ('Foobar Cv', 'Foobar$cv', NULL),
+  ('foobar de cv', 'foobar de$cv', 'de cv'),
+  ('Foobar De Cv', 'Foobar De$cv', 'De Cv'),
+  ('Foobar Gcv', 'Foobar Gcv', '~* \Scv$'),
+  ('Foobar Icv', 'Foobar Icv', 'Icv'),
+  ('Foobar Cvba', 'Foobar Cvba', 'Cvba'),
+  ('Cvfoo Bar', 'Cvfoo Bar',  '~* ^cv'),
+  ('Cv Foo Bar', 'Cv Foo Bar',  '~* ^cv\s'),
+  ('Foo Cvbar', 'Foo Cvbar',  '~* \scv\S'),
+  ('Foo Cv Bar', 'Foo Cv Bar',  '~* \scv\s'),
+  /* \s+dac$ */
+  ('foobar dac', 'foobar$dac', NULL),
+  ('Foobar Dac', 'Foobar$dac', NULL),
+  ('Foobar	Dac', 'Foobar$dac', 'tab whitespace'),
+  ('Big Bad Ac', 'Big Bad Ac', '''d ac'' does not replace on dac'),
+  ('Foodac', 'Foodac', NULL),
+  ('foo dac bar', 'foo dac bar', NULL),
+  /* \s+govt$ */
+  ('foobar govt', 'foobar$govt', NULL),
+  ('foobar Govt', 'foobar$govt', NULL),
+  ('foo govt dept of bar', 'foo govt dept of bar', NULL),
+  ('govt of foobar', 'govt of foobar', NULL),
+  ('foo intergovtl bar', 'foo intergovtl bar', 'intergovtl'),
+  ('foobar govts', 'foobar govts', NULL),
+  /* \s+hldgs$ */
+  ('foobar hldgs', 'foobar$hldgs', NULL),
+  ('foobar Hldgs', 'foobar$hldgs', NULL),
+  ('Foo Hldgs Bar', 'Foo Hldgs Bar',  'hldgs in middle of string not altered'),
+  ('foobar (Hldgs)', 'foobar (Hldgs)', 'hldgs in parens not altered'),
```

Comment:
we're not removing ( and ) ?

Reply:
This test is to show that if Hldgs isn't at the end of the string, it doesn't get parsed. An implicit behavior, but one that should be tested.